patch/driver-hwmon-pmbus-dps200.patch

```diff
@@ -0,0 +1,158 @@
++	    reg == PMBUS_VIN_OV_FAULT_LIMIT ||
++	    reg == PMBUS_VIN_OV_WARN_LIMIT ||
++	    reg == PMBUS_IIN_OC_FAULT_LIMIT ||
++	    reg == PMBUS_POUT_MAX)
++		return -ENXIO;
++
++	/*
++	 * NOTE: The following field have constant driver value.
++	 * If the register are PMBUS_IOUT_OC_WARN_LIMIT or
++	 * PMBUS_IOUT_OC_FAULT_LIMIT convert the constant value from linear
++	 * to direct format.
++	 */
++	if (reg == PMBUS_IOUT_OC_WARN_LIMIT)
++		return 0xb4;
++	else if (reg == PMBUS_IOUT_OC_FAULT_LIMIT)
++		return 0xc8;
++	else
++		return pmbus_read_word_data(client, page, reg);
++}
++
+ static struct pmbus_driver_info dps1900_info = {
+ 	.pages = 1,
+ 	.func[0] = PMBUS_HAVE_VIN | PMBUS_HAVE_VOUT | PMBUS_HAVE_STATUS_VOUT
+-	         | PMBUS_HAVE_IIN | PMBUS_HAVE_IOUT | PMBUS_HAVE_STATUS_IOUT
+-	         | PMBUS_HAVE_FAN12 | PMBUS_HAVE_STATUS_FAN12
+-	         | PMBUS_HAVE_TEMP | PMBUS_HAVE_TEMP2 | PMBUS_HAVE_STATUS_TEMP
+-	         | PMBUS_HAVE_STATUS_INPUT,
++		 | PMBUS_HAVE_IIN | PMBUS_HAVE_IOUT | PMBUS_HAVE_STATUS_IOUT
++		 | PMBUS_HAVE_FAN12 | PMBUS_HAVE_STATUS_FAN12
++		 | PMBUS_HAVE_TEMP | PMBUS_HAVE_TEMP2 | PMBUS_HAVE_STATUS_TEMP
++		 | PMBUS_HAVE_STATUS_INPUT,
+ 	.read_word_data = dps1900_read_word_data,
+ };
+ 
++/*
++ * From DPS-200 datasheet the supported sensors format defined as:
++ * VOUT: direct mode with one decimal place.
++ *
++ * Other sensors:
++ * IOUT: direct mode with one decimal place.
++ * IOUT Limits: linear mode, which difference from IOUT.
++ * VIN, IIN, POWER, TEMP, & FAN: linear mode.
++ */
++static struct pmbus_driver_info dps200_info = {
++	.pages = 1,
++	.func[0] = PMBUS_HAVE_VIN | PMBUS_HAVE_VOUT | PMBUS_HAVE_STATUS_VOUT
++		 | PMBUS_HAVE_IIN | PMBUS_HAVE_IOUT | PMBUS_HAVE_STATUS_IOUT
++		 | PMBUS_HAVE_PIN | PMBUS_HAVE_POUT
++		 | PMBUS_HAVE_FAN12 | PMBUS_HAVE_STATUS_FAN12
++		 | PMBUS_HAVE_TEMP | PMBUS_HAVE_TEMP2 | PMBUS_HAVE_TEMP3
++		 | PMBUS_HAVE_STATUS_TEMP | PMBUS_HAVE_STATUS_INPUT,
++	.format = {
++		[PSC_VOLTAGE_OUT] = direct,
++		[PSC_CURRENT_OUT] = direct,
++	},
++	.m = {
++		[PSC_VOLTAGE_OUT] = 10,
++		[PSC_CURRENT_OUT] = 10,
++	},
++	.b = {
++		[PSC_VOLTAGE_OUT] = 0,
++		[PSC_CURRENT_OUT] = 0,
++	},
++	.R = {
++		[PSC_VOLTAGE_OUT] = 0,
++		[PSC_CURRENT_OUT] = 0,
++	},
++	.read_word_data = dps200_read_word_data,
++};
++
+ static int dps1900_probe(struct i2c_client *client,
+ 			 const struct i2c_device_id *id)
+ {
+-	return pmbus_do_probe(client, id, &dps1900_info);
++	struct pmbus_driver_info *info;
++
++	info = devm_kzalloc(&client->dev, sizeof(struct pmbus_driver_info),
++			    GFP_KERNEL);
++	if (!info)
++		return -ENOMEM;
++
++	switch (id->driver_data) {
++	case dps1900:
++		info = &dps1900_info;
++		break;
++	case dps200:
++		info = &dps200_info;
++		break;
++	default:
++		dev_err(&client->dev, "Unsupported device\n");
++		return -ENODEV;
++	}
++
++	return pmbus_do_probe(client, id, info);
+ }
+ 
+ static const struct i2c_device_id dps1900_id[] = {
+-	{"dps1900", 0},
```

Comment:
does this id change affect dps1900 driver?

Reply:
@lguohan It does not affect the dps1900 driver.